ethereum/contracts/ShutdownSwitch.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.0;
+
+import "./interfaces/IWormhole.sol";
+
+contract ShutdownStructs {
+    struct VoterEntry {
+        address key;
+        bool enabled;
+    }
+}
+
+abstract contract ShutdownSwitch {
+    function requiredVotesToDisable() public view returns (uint16) {
+        return requiredNoVotes;
+    }
+
+    function numVotesToDisable() public view returns (uint16) {
+        return votesToDisable;
+    }
+
+    function enabledFlag() public view returns (bool){
+        return enabled;
+    }
+
+    event ShutdownVoteCast(address indexed voter, bool votedToEnable, uint16 numVotesToDisable, bool enableFlag);
+    event ShutdownStatusChanged(bool enabledFlag, uint16 numVotesToDisable);
+
+    function getWH() public virtual view returns (IWormhole);
```

Comment:
I think this leaks out too much internal details. In principle, the emergency shutdown class isn't tied to wormhole at all, so instead of asking for the whole `IWormhole`, we could just have a
```solidity
function eligibleVoters() public virtual view returns address[];
```
which could be implemented by returning the current guardian set.

Reply:
So it `eligibleVoters()` would be something to this effect?

```
Structs.GuardianSet memory guardianSet = getGuardianSet(getCurrentGuardianSetIndex());
return guardianSet.keys
```

Reply:
Yes, that's what I had in mind.

Reply:
Did this.